### PR TITLE
Jd/dpc 4176 cpi profile endpoint

### PR DESCRIPTION
**Pull request by jdettmannnava**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4176

## 🛠 Changes

- Added call to profile endpoint to CpiApiGatewayClient
- Removed call to roles endpoint from CpiApiGatewayClient
- Replace multiple calls for verifying AO in AoVerificationService with call to new endpoint

## ℹ️ Context

The CPI API Gateway team added a new endpoint that consolidates the information we were getting from the enrollments and enrollments/[id]/roles endpoints. This change is potentially much more efficient. [Excruciating detail here](https://confluence.cms.gov/x/o2A2Q).

Note: I did not remove the call to enrollments, as it is used in the VerifyProviderOrganizationJob.

## 🧪 Validation

Manual and Automated tests.
Called the CPI API Gateway VAL environment and received the documented response.

### CODE REVIEW

**Review by MEspositoE14s — APPROVED** [review]
Everything works when I run locally 👍

File: dpc-portal/app/services/ao_verification_service.rb [hunk 1]
```diff
@@ -82,11 +80,19 @@ def waiver?(waivers_list)
...
+    response = @cpi_api_gw_client.fetch_profile(organization_npi)
+    raise AoException, 'bad_npi' if response['code'] == '404'
+
+    enrollments = response.dig('provider', 'enrollments')&.select { |enrollment| enrollment['status'] == 'APPROVED' }
+    raise AoException, 'no_approved_enrollment' if enrollments.blank?
+
+    role_from_enrollments(enrollments, identifier_type, identifier)
+  end
+
+  def role_from_enrollments(enrollments, identifier_type, identifier)
```
Comment (chris-ronning-ny) [inline dpc-portal/app/services/ao_verification_service.rb:93]: Maybe `check_role_from_enrollments`? This method name isn't super clear.

**Review by chris-ronning-ny — COMMENTED** [review]
This looks good, and I'm glad we can consolidate so much! 

One question: Do we still need `ao_verification_service.get_approved_enrollments` and `cpi_api_gw_client.fetch_enrollment` to exist? If I understand correctly, the results from `fetch_enrollment` are consolidated into the new profile endpoint, and `get_approved_enrollments` just runs the same checks we're currently running.

File: dpc-portal/app/services/ao_verification_service.rb (hunk 1 above)
Comment (jdettmannnava) [reply dpc-portal/app/services/ao_verification_service.rb:93]: Well, it actually returns the role that matches the parameters

**Comment by jdettmannnava** [comment]
> One question: Do we still need `ao_verification_service.get_approved_enrollments` and `cpi_api_gw_client.fetch_enrollment` to exist? If I understand correctly, the results from `fetch_enrollment` are consolidated into the new profile endpoint, and `get_approved_enrollments` just runs the same checks we're currently running.

Yes, we still need it for the weekly verification checks to make sure that organizations that do not have an active AO still have an approved enrollment

**Review by chris-ronning-ny — APPROVED** [review]

**Review by MEspositoE14s — APPROVED** [review]
Looks good to me!

**Pull request merged**